Plugin Dependencies: Remove auto-deactivation and bootstrapping logic.

costdev opened the pull request:
Automatic-deactivation of dependents with unmet dependencies requires a write operation to the database. This was performed in Core's bootstrap, and risked the database and cache becoming out-of-sync on sites with heavy traffic.

No longer loading plugins that have unmet requirements has not had a final approach decided core-wide, and is still in discussion in another ticket that will be handled in the future.

The `plugin_data` option, used to persistently store plugin data for detecting unmet dependencies in the bootstrap, is no longer needed.

Trac ticket: https://core.trac.wordpress.org/ticket/60457

costdev committed:
Remove bootstrapping.

costdev committed:
Remove auto-deactivation.

costdev committed:
Remove `plugin_data` option and logic.

costdev committed:
Add guards against uninitialized data.

costdev committed:
Initialize Plugin Dependencies on plugin pages.

costdev committed:
Initialize Plugin Dependencies in `validate_plugin_requirements()`.

costdev committed:
Guard initialization with a new `bool` property.

costdev committed:
Tests: Add test to ensure initialization runs only once.

costdev committed:
Ensure that `active_plugins` returns an array if not present.

costdev committed:
Tests: Add tests for uninitialized data guards.

swissspidy commented:
```
There were 3 failures:

1) Tests_Multisite_Network::test_active_network_plugins
Failed asserting that two arrays are identical.
--- Expected
+++ Actual
@@ @@
-Array &0 (
-    0 => '/var/www/tests/phpunit/data/plugins/hello.php'
-)
+Array &0 ()

/var/www/tests/phpunit/tests/multisite/network.php:221
phpvfscomposer:///var/www/vendor/phpunit/phpunit/phpunit:106
/var/www/vendor/bin/phpunit:118

2) Tests_Multisite_Network::test_duplicate_network_active_plugin
Failed asserting that actual size 0 matches expected size 1.

/var/www/tests/phpunit/tests/multisite/network.php:247
phpvfscomposer:///var/www/vendor/phpunit/phpunit/phpunit:106
/var/www/vendor/bin/phpunit:118

3) Tests_Multisite_Network::test_is_plugin_active_for_network_true
Failed asserting that false is true.

/var/www/tests/phpunit/tests/multisite/network.php:261
phpvfscomposer:///var/www/vendor/phpunit/phpunit/phpunit:106
/var/www/vendor/bin/phpunit:118
```

costdev committed:
Tests: Change `set_up()` to `tear_down()`.

Woops. I even wrote the docblock to say "after each test" yet I used `set_up()`.

Brain did not work well that day.

costdev committed:
Docs: Update the docblock for `::initialize()`.

costdev commented:
Committed in [r57658](https://core.trac.wordpress.org/changeset/57658).
